### ISSUE TITLE
Add episode `guid` front matter param to all episode

### ISSUE THREAD

**Issue by kbondarev** [JupiterBroadcasting/jupiterbroadcasting.com#93]
This would be required to pass as a URL param to the podverse player (see #20) 

Need to modify the scraper to make it add a `guid` param to the episode's front matter json and then add the changes to all the episode files.

The scraper already looks at the fireside json api format of each show (e.g. coder.show/json), the `guid` could be grabbed from there. It is the `id` param of each episode in `items` array.

**Comment by StefanS-O** [comment]
But how about the future? We need to generate an ID in Hugo.
I suggest something like this: `{{ md5 .Permalink }} ` which outputs this: `0afbeb4e1e28bf94e7e21af72bf90f11 `, no?
So when we have no GUID from source (like Fireside / Wordpress), we fall back on generating our own?

**Comment by kbondarev** [comment]
Yes. In the future would have to make our own.

md5 of the permalink would be fine, since it encapsulates the show and episode number in it.

**Comment by kbondarev** [comment]
Hmm maybe md5 is not correct actually.

this is the guid doc for the podcast (the whole show, not specific episode. I couldn’t find the one for the item - the episode, but I imagine it’s similar):
https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#guid

So it looks like it needs to be a UUIDv5

**Comment by kbondarev** [comment]
More info on UUIDv5 on this [page](https://www.sohamkamani.com/uuid-versions-explained/)

I think the “namespace” should be the uuid of the whole show and the “input string” just the episode number as an int.

**Comment by StefanS-O** [comment]
I am just not sure if we can generate it using Hugo, that would suck.

**Comment by kbondarev** [comment]
Is there a way to use go dependencies (third-party libs) during build?
I found this: https://github.com/google/uuid

Or any ideas connecting something external for this? For example, I could super quickly bang a little python thing that would generate this, it could be a REST API or a CLI tool, or whatever. The question is if it could be somehow integrated with hugo/frontmatter

**Comment by kbondarev** [comment]
I think when we move aways from
Fireside we should keep the same
GUID for all the past shows. And generate only the new ones ourselves.

As I understand  guid of the episode is important for the podcast apps. That’s how they know for example if that episode was played already or not. If we change the old ones, it might mess things up for ppl.

**Comment by mitchdowney** [comment]
@kbondarev yes to your point, guids allow podcast apps to know which <item> should be treated as the same <item> in the future, even if much of the other information for that <item> changes over time. Without a GUID, the next best "unique identifier" we have is in the enclosure URL, but that can change over time for the same episode.

One quick note...Podverse currently does not parse the <podcast:guid> namespace. We do however parse the older <guid> namespace. Unless you have objections, I would recommend including both <guid> and <podcast:guid> as identical values, but I'm really not familiar enough with the purpose of the <podcast:guid> namespace vs <guid> so maybe my suggestion is a bad idea.

If it's necessary for <guid> and <podcast:guid> to be different values, then that will prompt us to add <podcast:guid> support to Podverse sooner, although soonest I could do that is this weekend.

**Comment by kbondarev** [comment]
@mitchdowney, I appreciate a lot your quick fix and I appologize for the late feedback. It's been a busy week.

I implmented everything as you describe it and it works :smile: :fire: 
However, there are a few rough edges/pain points:

1) The player ends up rendering in the  ["show-all-episodes" mode](https://podverse.fm/embed/player?podcastId=g40Um-HP1&episodeGuid=92ab2f36-1fef-4cbc-8cec-87b245161c1e&episodeMediaUrl=https://aphid.fireside.fm/d/1437767933/f31a453c-fa15-491f-8618-3f71f1d565e5/92ab2f36-1fef-4cbc-8cec-87b245161c1e.mp3&episodeTitle=custom%20title&episodePubDate=800000000) rather than the ["only-one-episode" mode](https://podverse.fm/embed/player?episodeId=kilYJsW3A) which is produced when give only the `episodeId`.

This not ideal, but also not a big issue since the iframe hides it:

<kbd>![image](https://user-images.githubusercontent.com/7744333/179157434-bf5efdef-f511-4721-82f9-58676b441e2b.png)</kbd>

However, the "show-all-episodes" mode doesn't hide the "full-screen" button at the top-right corner of the player. It becomes a mess if you click it, since it's inside the iframe:

<kbd>![image](https://user-images.githubusercontent.com/7744333/179157886-724677d2-e752-4606-93cd-20d86882240a.png)</kbd>


Is it possible to send more events to the parent like [here](https://github.com/podverse/podverse-web/blob/develop/pages/embed/player.tsx#L59).
For example, when the `full-screen` mode is toggled on/off. I could maybe then change the style of the `<iframe>` itself and make it also cover the whole screen. It might be a neat hack around it :)
Then we would not lose those extra control buttons in the full-screen mode.
I don't know how the other events for user actions might become handy, but I don't see any reason why not send them :). For example, clicked-play, clicked-pause, clicked-next-chapter, etc.



I looked into the code a little, and found the `showAllEpisodes` boolean URL param that I could pass, but it didn't have an effect.


2) I also wanted to change the style of the player to match better the theme on the site. I looked at [this example](https://podverse.fm/embed/player-demo-custom-css) and [the code](https://github.com/podverse/podverse-web/blob/develop/pages/embed/player-demo-custom-css.tsx) and implemented something similar. But it results in a CORS error:

```
Uncaught DOMException: Blocked a frame with origin "http://localhost:1313" from accessing a cross-origin frame.
    at pvSetCSSVariable (http://localhost:1313/show/linux-unplugged/466/:423:41)
    at pvHandleCSSOverrides (http://localhost:1313/show/linux-unplugged/466/:426:13)
```

Maybe the way to do this is to pass the CSS as URL query params as well?

**Comment by kbondarev** [comment]
Closed by #105
